cv/lane_detection/src/lane_detection_model_inference.py

```diff
@@ -1,136 +1,143 @@
     height = img.shape[0]
 
-    gray_im = cv2.cvtColor(img,cv2.COLOR_BGR2GRAY)
+    gray_im = cv2.cvtColor(img, cv2.COLOR_BGR2GRAY)
     # gray_im = cv2.GaussianBlur(gray_im,(3,3),0)
     # Separate into quadrants
-    med1 = np.median(gray_im[:height//2,:width//2])
-    med2 = np.median(gray_im[:height//2,width//2:])
-    med3 = np.median(gray_im[height//2:,width//2:])
-    med4 = np.median(gray_im[height//2:,:width//2])
+    med1 = np.median(gray_im[: height // 2, : width // 2])
+    med2 = np.median(gray_im[: height // 2, width // 2 :])
+    med3 = np.median(gray_im[height // 2 :, width // 2 :])
+    med4 = np.median(gray_im[height // 2 :, : width // 2])
 
-    l1 = int(max(0,(1-0.205)*med1))
-    u1 = int(min(255,(1+0.205)*med1))
-    e1 = cv2.Canny(gray_im[:height//2,:width//2],l1,u1)
+    l1 = int(max(0, (1 - 0.205) * med1))
+    u1 = int(min(255, (1 + 0.205) * med1))
+    e1 = cv2.Canny(gray_im[: height // 2, : width // 2], l1, u1)
 
-    l2 = int(max(0,(1-0.205)*med2))
-    u2 = int(min(255,(1+0.205)*med2))
-    e2 = cv2.Canny(gray_im[:height//2,width//2:],l2,u2)
+    l2 = int(max(0, (1 - 0.205) * med2))
+    u2 = int(min(255, (1 + 0.205) * med2))
+    e2 = cv2.Canny(gray_im[: height // 2, width // 2 :], l2, u2)
 
-    l3 = int(max(0,(1-0.205)*med3))
-    u3 = int(min(255,(1+0.205)*med3))
-    e3 = cv2.Canny(gray_im[height//2:,width//2:],l3,u3)
+    l3 = int(max(0, (1 - 0.205) * med3))
+    u3 = int(min(255, (1 + 0.205) * med3))
+    e3 = cv2.Canny(gray_im[height // 2 :, width // 2 :], l3, u3)
 
-    l4 = int(max(0,(1-0.205)*med4))
-    u4 = int(min(255,(1+0.205)*med4))
-    e4 = cv2.Canny(gray_im[height//2:,:width//2],l4,u4)
+    l4 = int(max(0, (1 - 0.205) * med4))
+    u4 = int(min(255, (1 + 0.205) * med4))
+    e4 = cv2.Canny(gray_im[height // 2 :, : width // 2], l4, u4)
 
     # Stitch the edges together
-    edges_mask[:height//2,:width//2] = e1
-    edges_mask[:height//2,width//2:] = e2
-    edges_mask[height//2:,width//2:] = e3
-    edges_mask[height//2:,:width//2] = e4
+    edges_mask[: height // 2, : width // 2] = e1
+    edges_mask[: height // 2, width // 2 :] = e2
+    edges_mask[height // 2 :, width // 2 :] = e3
+    edges_mask[height // 2 :, : width // 2] = e4
 
     edges_mask_inv = cv2.bitwise_not(edges_mask)
 
     return edges_mask, edges_mask_inv
 
 
-def get_input(frame):
-    frame = cv2.resize(frame,(1280,720),interpolation=cv2.INTER_AREA)
+def get_input(frame: np.ndarray) -> np.ndarray:
+    frame = cv2.resize(frame, (330, 180), interpolation=cv2.INTER_AREA)
     frame_copy = np.copy(frame)
 
-    test_edges,test_edges_inv = find_edge_channel(frame_copy)
-    frame_copy = np.append(frame_copy,test_edges.reshape(test_edges.shape[0],test_edges.shape[1],1),axis=2)
-    frame_copy = np.append(frame_copy,test_edges_inv.reshape(test_edges_inv.shape[0],test_edges_inv.shape[1],1),axis=2)
-    frame_copy = cv2.resize(frame_copy,(330,180))
+    test_edges, test_edges_inv = find_edge_channel(frame_copy)
```

Comment:
REAL CHANGE

I combined the np.append's into a single np.block. I also moved the resize to the top (I'm fairly sure the function that is called on frame_copy is shape agnostic)